libs/cli/src/main/java/org/elasticsearch/cli/Terminal.java

```diff
@@ -158,6 +166,22 @@ private static PrintWriter newWriter() {
             return new PrintWriter(System.out);
         }
 
+        private BufferedReader getReader() {
+            if (reader == null) {
+                reader = new BufferedReader(new InputStreamReader(System.in, Charset.defaultCharset()));
+            }
+            return reader;
+        }
+
+        /** Visible for testing the system terminal with custom input */
+        @Override
+        protected void setInput(InputStream inputStream) throws IOException {
```

Comment:
Instead of having this method just for testing on the actual Terminal class, I think it could be on MockTerminal? We could have a MockSystemTerminal if necessary.  I especially don't like the close here; we shouldn't ever need to close the reader, we should just create a new MockTerminal with different input.

Reply:
I think I can get rid of this altogether.